lib/spack/spack/modules/common.py

```diff
@@ -428,12 +428,17 @@ class BaseConfiguration(object):
 
     default_projections = {"all": "{name}-{version}-{compiler.name}-{compiler.version}"}
 
-    def __init__(self, spec, module_set_name):
+    def __init__(self, spec, module_set_name, explicit=None):
         # Module where type(self) is defined
         self.module = inspect.getmodule(self)
         # Spec for which we want to generate a module file
         self.spec = spec
         self.name = module_set_name
+        # Software installation has been explicitly asked (get this
+        # information from spec if not provided as argument)
+        if explicit is None:
```

Comment:
Under what conditions is this branch followed?

Reply:
When we query existing modulefiles with `get_module` def (defined in [lib/spack/spack/modules/common.py](https://github.com/spack/spack/blob/develop/lib/spack/spack/modules/common.py#L400-L403)), we need to query the database to learn from it if the module is implicit or explicit.

I will update the comment to clarify.